aggregator/src/app/HealthService.ts

```diff
@@ -0,0 +1,192 @@
+import { axiod } from "../../deps.ts";
+import BundleTable from "./BundleTable.ts";
+import AppEvent from "./AppEvent.ts";
+import assert from "../helpers/assert.ts";
+import * as env from "../env.ts";
+
+abstract class HealthIndicator {
+  constructor(
+    public emit: (evt: AppEvent) => void,
+  ){}
+  abstract name: string;
+  status: ResourceHealth = ResourceHealth.Unhealthy;
+  details: string | undefined;
+  abstract checkHealth(): Promise<void> | void;
+}
+
+export enum ResourceHealth {
+  Healthy = 'HEALTHY',
+  Unhealthy = 'UNHEALTHY'
+}
```

Comment:
We don't really use enums... I suggest this should be replaced with something like:

```ts
export type ResourceHealth = "healthy" | "unhealthy";
```

Open to discussing this further, especially if you have a concrete reason for preferring the enum version for these changes.